src/test/java/com/salesforce/storm/spout/sideline/trigger/example/ZookeeperWatchTriggerTest.java

```diff
@@ -0,0 +1,335 @@
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test that the reference implementation for sideline triggers works correctly.
+ */
+public class ZookeeperWatchTriggerTest {
+
+    private static final String CREATED_BY = "Test";
+    private static final String DESCRIPTION = "Description";
+
+    /**
+     * Create shared zookeeper test server.
+     */
+    @ClassRule
+    public static final SharedZookeeperTestResource sharedZookeeperTestResource = new SharedZookeeperTestResource();
+
+    /**
+     * Test that the sideline trigger opens and that it calls start and stop properly when nodes are changed.
+     * @throws Exception something bad
+     */
+    @Test
+    public void testOpenWithNewEvents() throws Exception {
+        final String zkRoot = "/test-trigger" + System.currentTimeMillis();
+
+        final String consumerId = "VirtualSpoutPrefix";
+
+        final Map<String, Object> config = SpoutConfig.setDefaults(new HashMap<>());
+        config.put(SpoutConfig.VIRTUAL_SPOUT_ID_PREFIX, consumerId);
+        config.put(Config.ZK_SERVERS, Collections.singletonList(getZkServer().getConnectString()));
+        config.put(Config.ZK_ROOTS, Collections.singletonList(zkRoot));
+        config.put(SpoutConfig.CONSUMER_CLASS, MockConsumer.class.getName());
+        config.put(
+            SpoutConfig.PERSISTENCE_ADAPTER_CLASS,
+            com.salesforce.storm.spout.dynamic.persistence.InMemoryPersistenceAdapter.class.getName()
+        );
+        config.put(
+            SidelineConfig.PERSISTENCE_ADAPTER_CLASS,
+            InMemoryPersistenceAdapter.class.getName()
+        );
+        config.put(Config.FILTER_CHAIN_STEP_BUILDER_CLASS, MockFilterChainStepBuilder.class.getName());
+
+        final CuratorFramework currator = CuratorFactory.createNewCuratorInstance(
+            Tools.stripKeyPrefix(Config.PREFIX, config)
+        );
+        final CuratorHelper curatorHelper = new CuratorHelper(currator);
+
+        final SidelineSpoutHandler sidelineSpoutHandler = Mockito.mock(SidelineSpoutHandler.class);
+
+        final ZookeeperWatchTrigger trigger = new ZookeeperWatchTrigger();
+        trigger.setSidelineController(sidelineSpoutHandler);
+        trigger.open(config);
+
+        // We're going to turn some events into JSON
+        final Gson gson = new GsonBuilder()
+            .setDateFormat("yyyy-MM-dd HH:mm:ss")
+            .create();
+
+        final Map<String,Object> startData = new HashMap<>();
+        startData.put("id", "start");
+        final TriggerEvent startTriggerEvent = new TriggerEvent(
+            SidelineType.START,
+            startData,
+            new Date(),
+            CREATED_BY,
+            DESCRIPTION,
+            false,
+            new Date()
```

Comment:
Do we really wanna use java.util.Date?  Its pretty awful :/

Reply:
What do you want to use? I thought you stripped joda out?

Reply:
java.util.Date just has the world worst API.  Maybe the new Java8 replacement for it?

http://www.baeldung.com/java-8-date-time-intro  using GMT/UTC timezone of course :p

Reply:
Switched out for LocalDateTime. We don't actually use these for computations, so I'm not sure how much we care about the timezone. If you think these all need to be Zoned let me know and I'll pick it up in a subsequent PR.